validator/src/admin_rpc_service.rs

```diff
@@ -267,6 +290,30 @@ impl AdminRpc for AdminRpcImpl {
         Ok(())
     }
 
+    fn set_block_engine_config(
+        &self,
+        meta: Self::Metadata,
+        block_engine_url: String,
+        trust_packets: bool,
+    ) -> Result<()> {
+        debug!("set_block_engine_config request received");
+
+        if block_engine_url.contains("http") || block_engine_url.is_empty() {
+            meta.with_post_init(|post_init| {
+                *post_init.block_engine_config.lock().unwrap() = BlockEngineConfig {
+                    block_engine_url,
+                    trust_packets,
+                };
+                Ok(())
+            })
+        } else {
+            Err(jsonrpc_core::error::Error::invalid_params(format!(
+                "invalid block_engine_url: {} - must point to an http(s) connection or empty string.", 
```

Comment:
https case goes to this error?

Reply:
no, that contains "http", but this is replaced with a more robust validate function on 1.14 branch, will be here soon